pilot/pkg/xds/endpoint_builder.go

```diff
@@ -282,7 +282,7 @@ func (b *EndpointBuilder) buildLocalityLbEndpointsFromShards(
 			if !ep.IsDiscoverableFromProxy(b.proxy) {
 				continue
 			}
-			if svcPort.Name != ep.ServicePortName {
+			if svcPort.Port != int(ep.EndpointPort) {
```

Comment:
This is not right, svc port can be different from endpoint port

Reply:
yes your are right。 i test find that

Reply:
Do you have any idea to fix the bug？

Reply:
This is a good catcha, will investigate

Reply:
i find a way to fix the bug？ still hava 503 for a little while .send next xds is ok.